x-pack/plugins/alerting/server/rules_client/methods/aggregate.ts

```diff
@@ -25,13 +26,14 @@ export async function aggregate<T = Record<string, unknown>>(
   params: AggregateParams<T>
 ): Promise<T> {
   const { options = {}, aggs } = params;
-  const { filter, page = 1, perPage = 0, ...restOptions } = options;
+  const { filter, page = 1, perPage = 0, filterConsumers, ...restOptions } = options;
 
   let authorizationTuple;
   try {
     authorizationTuple = await context.authorization.getFindAuthorizationFilter(
       AlertingAuthorizationEntity.Rule,
-      alertingAuthorizationFilterOpts
+      alertingAuthorizationFilterOpts,
+      isEmpty(filterConsumers) ? undefined : new Set(filterConsumers)
```

Comment:
Does this respect RBAC? API users shouldn't be able to pass whatever set of consumers they want to aggregate over, they should just be able to limit which consumers to aggregate over that they already have access to correct?

Reply:
Yes it does follow RBAC, thanks for scaring me here ;) but we do, we are just saying that we only care some consumers. It is helping us to only show o11y rules in the rule management page.